### DISCUSSION TITLE
How to seperate redis.op by redis.cmd

### DISCUSSION THREAD

**Discussion by Jedore** [apache/skywalking#10211] [Q&A] [answered]
In python agent plugin, there is no implementation for virtual cache of Redis. I want to add it. So I want to know how to  seperate `redis.op` by `redis.cmd`?

https://skywalking.apache.org/docs/main/v9.3.0/en/setup/service-agent/virtual-cache/

**Comment by wu-sheng** [comment, marked as answer]
> So I want to know how to seperate redis.op by redis.cmd

There is no magic here, change the source codes of the plugin, and follow the `read`/`write` command catalog.

Ref Java agent implementation, https://github.com/apache/skywalking-java/blob/main/apm-sniffer/apm-sdk-plugin/jedis-plugins/jedis-4.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/jedis/v4/JedisPluginConfig.java#L52

**Reply by Jedore** [reply]
Ok, thanks.
